src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/integration/VideoLinkMigrateIntegrationTest.kt

```diff
@@ -0,0 +1,259 @@
+package uk.gov.justice.digital.hmpps.whereabouts.integration
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.jdbc.core.JdbcTemplate
+import org.springframework.test.context.transaction.TestTransaction
+import org.springframework.test.jdbc.JdbcTestUtils
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.whereabouts.dto.VideoBookingMigrateResponse
+import java.time.LocalDateTime
+
+@Transactional
+class VideoLinkMigrateIntegrationTest : IntegrationTest() {
+  @Autowired
+  lateinit var jdbcTemplate: JdbcTemplate
+
+  @BeforeEach
+  fun prepare() {
+    deleteAllTestRows()
+    TestTransaction.flagForCommit()
+    TestTransaction.end()
+  }
+
+  @Test
+  fun `Will return a simple booking to migrate`() {
+    makeABooking()
+    val result = webTestClient.get()
+      .uri("/migrate/video-link-booking/1")
+      .headers {
+        it.setBearerAuth(
+          jwtAuthHelper.createJwt(
+            subject = "TEST-USER",
+            roles = listOf("ROLE_BOOK_A_VIDEO_LINK_ADMIN"),
+            clientId = "hmpps-book-a-video-link-client",
+          ),
+        )
+      }
+      .exchange()
+      .expectStatus().isOk
+      .expectBody(VideoBookingMigrateResponse::class.java)
+      .returnResult()
+      .responseBody
+
+    assertThat(result).isNotNull
+
+    with(result!!) {
+      assertThat(cancelled).isFalse
+      assertThat(events.size).isEqualTo(1)
+      assertThat(courtCode).isEqualTo("CVNTCC")
+      assertThat(madeByTheCourt).isTrue
+      assertThat(probation).isFalse
+      assertThat(prisonCode).isEqualTo("MDI")
+
+      // This is populated from the appointment data - 2023-11-10 10:30-11:00am
+      with(main) {
+        assertThat(locationId).isEqualTo(1)
+        assertThat(date).isEqualTo("2023-11-10")
+        assertThat(startTime).isEqualTo("10:30")
+        assertThat(endTime).isEqualTo("11:00")
+      }
+
+      assertThat(result.pre).isNull()
+      assertThat(result.post).isNull()
+    }
+  }
+
+  @Test
+  fun `Will return a cancelled booking to migrate`() {
+    makeABooking(cancelled = true)
+    val result = webTestClient.get()
+      .uri("/migrate/video-link-booking/1")
+      .headers {
+        it.setBearerAuth(
+          jwtAuthHelper.createJwt(
+            subject = "TEST-USER",
+            roles = listOf("ROLE_BOOK_A_VIDEO_LINK_ADMIN"),
+            clientId = "hmpps-book-a-video-link-client",
+          ),
+        )
+      }
+      .exchange()
+      .expectStatus().isOk
+      .expectBody(VideoBookingMigrateResponse::class.java)
+      .returnResult()
+      .responseBody
+
+    assertThat(result).isNotNull
+
+    with(result!!) {
+      assertThat(cancelled).isTrue
+      assertThat(events.size).isEqualTo(2)
```

Comment:
A few assertions around the events might be useful, order and values etc?

Reply:
Yes agreed. Added a couple of checks here